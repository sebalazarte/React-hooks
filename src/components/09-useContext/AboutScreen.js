import React, { useContext } from 'react'
import { UserContext } from './UserContext'

export const AboutScreen = () => {

    const userContext = useContext(UserContext);
    const { user, setUser } = userContext;

    const handleClick = () => {
        setUser({});
    }

    return (
        <div>
            <h1>About Screen</h1>
            <hr />
            <pre className="blockquote">
                {JSON.stringify(user, null, 3)}
            </pre>
            <button
                className="btn btn-outline-primary"
                type="button"
                onClick={handleClick}
            >Logout</button>
        </div>
    )
}
