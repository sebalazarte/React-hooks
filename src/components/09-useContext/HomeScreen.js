import React, {useContext} from 'react'
import { UserContext } from './UserContext'

export const HomeScreen = () => {

    const userContext = useContext(UserContext);
    const {user} = userContext;

    return (
        <div>
            <h1>Home Screen</h1>
            <hr/>
            <pre className="blockquote">
                {JSON.stringify(user, null, 3)}
            </pre>
            
        </div>
    )
}
