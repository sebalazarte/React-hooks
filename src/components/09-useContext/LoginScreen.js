import React, { useContext } from 'react';
import { UserContext } from './UserContext'

export const LoginScreen = () => {

    const userContext = useContext(UserContext);
    const { user, setUser } = userContext;

    const handleClick = () => {

        const newUser = {
            id: 123,
            name: 'Sebastian',
            email: 'sebalazarte@outlook.com'
        };

        setUser(newUser);
    }

    return (
        <div>
            <h1>Login Screen</h1>
            <hr />
            <button
                className="btn btn-outline-primary"
                type="button"
                onClick={handleClick}
            >Login</button>

            <pre className="blockquote">
                {JSON.stringify(user, null, 3)}
            </pre>

        </div>
    )
}
