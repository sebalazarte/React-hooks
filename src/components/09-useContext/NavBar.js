import React from 'react'
import { NavLink } from 'react-router-dom';

export const NavBar = () => {
    return (
        <nav className="navbar navbar-expand-sm navbar-dark bg-dark">
          <NavLink className="navbar-brand" to="/">
            useContext
          </NavLink>
          <button className="navbar-toggler" type="button" data-toggle="collapse" data-target="#navbarText" aria-controls="navbarText" aria-expanded="false" aria-label="Toggle navigation">
            <span className="navbar-toggler-icon" />
           </button>
                <div className="collapse navbar-collapse" id="navbarText">
                  <ul className="navbar-nav mr-auto">
                    <li className="nav-item">
                      <NavLink 
                        exact activeClassName="active" 
                        className="nav-link" 
                        to="/">
                        Home
                      </NavLink>
                    </li>
                    <li className="nav-item">
                        <NavLink 
                            exact 
                            activeClassName="active" 
                            className="nav-link" 
                            to="/about">
                        About
                      </NavLink>
                    </li>
                  </ul>
                <NavLink 
                    exact 
                    activeClassName="active" 
                    className="btn btn-outline-info my-2 my-sm-0" 
                    to="/login">
                    Login
                  </NavLink>
                  <NavLink
                    exact 
                    activeClassName="active"
                    className="btn btn-outline-info my-2 my-sm-0 ml-3"
                    to="/register"
                  >
                    Register
                  </NavLink>
                </div>
              </nav>
        
    );
}
