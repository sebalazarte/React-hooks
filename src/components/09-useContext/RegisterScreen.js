import React from 'react'

export const RegisterScreen = () => {
    return (
        <div>
            <h1>Register screen</h1>
            <hr/>
        </div>
    )
}
