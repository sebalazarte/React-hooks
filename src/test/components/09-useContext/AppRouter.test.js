import { mount } from 'enzyme';
import React from 'react';
import { AppRouter } from '../../../components/09-useContext/AppRouter';
import { UserContext } from '../../../components/09-useContext/UserContext';

describe('AppRouter', () => {

    const user = {
        id: 123,
        name: 'Sebastian',
        email: 'sebalazarte@outlook.com'
    };

    const wrapper = mount(
        <UserContext.Provider value={{user}}>
            <AppRouter />
        </UserContext.Provider>
    );

    test('debe mostrarse correctamente', () => {
        expect(wrapper).toMatchSnapshot();
    });

});