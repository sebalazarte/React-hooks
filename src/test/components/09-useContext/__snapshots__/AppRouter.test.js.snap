// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppRouter debe mostrarse correctamente 1`] = `
<div>
  <nav
    className="navbar navbar-expand-sm navbar-dark bg-dark"
  >
    <a
      aria-current="page"
      className="navbar-brand active"
      href="/"
      onClick={[Function]}
      style={Object {}}
    >
      useContext
    </a>
    <button
      aria-controls="navbarText"
      aria-expanded="false"
      aria-label="Toggle navigation"
      className="navbar-toggler"
      data-target="#navbarText"
      data-toggle="collapse"
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      className="collapse navbar-collapse"
      id="navbarText"
    >
      <ul
        className="navbar-nav mr-auto"
      >
        <li
          className="nav-item"
        >
          <a
            aria-current="page"
            className="nav-link active"
            href="/"
            onClick={[Function]}
            style={Object {}}
          >
            Home
          </a>
        </li>
        <li
          className="nav-item"
        >
          <a
            aria-current={null}
            className="nav-link"
            href="/about"
            onClick={[Function]}
          >
            About
          </a>
        </li>
      </ul>
      <a
        aria-current={null}
        className="btn btn-outline-info my-2 my-sm-0"
        href="/login"
        onClick={[Function]}
      >
        Login
      </a>
      <a
        aria-current={null}
        className="btn btn-outline-info my-2 my-sm-0 ml-3"
        href="/register"
        onClick={[Function]}
      >
        Register
      </a>
    </div>
  </nav>
  <div
    className="container"
  >
    <div>
      <h1>
        Home Screen
      </h1>
      <hr />
      <pre
        className="blockquote"
      >
        {
   "id": 123,
   "name": "Sebastian",
   "email": "sebalazarte@outlook.com"
}
      </pre>
    </div>
  </div>
</div>
`;
